ext/standard/array.c

```diff
@@ -681,63 +1017,141 @@ PHP_FUNCTION(natcasesort)
 
 typedef bucket_compare_func_t(*get_compare_function)(zend_long);
 
-static zend_always_inline void php_sort(INTERNAL_FUNCTION_PARAMETERS, get_compare_function get_cmp, bool renumber) {
+static zend_always_inline void php_array_apply_sort(HashTable *array, zend_long sort_type, get_compare_function get_cmp, bool renumber)
+{
+	bucket_compare_func_t cmp = get_cmp(sort_type);
+	zend_array_sort(array, cmp, renumber);
+}
+
+/* {{{ Sort an array and maintain index association */
+PHP_FUNCTION(asort)
+{
 	HashTable *array;
 	zend_long sort_type = PHP_SORT_REGULAR;
-	bucket_compare_func_t cmp;
 
 	ZEND_PARSE_PARAMETERS_START(1, 2)
 		Z_PARAM_ARRAY_HT_EX(array, 0, 1)
 		Z_PARAM_OPTIONAL
 		Z_PARAM_LONG(sort_type)
 	ZEND_PARSE_PARAMETERS_END();
 
-	cmp = get_cmp(sort_type);
-
-	zend_array_sort(array, cmp, renumber);
+	if ((sort_type & ~PHP_SORT_FLAG_CASE) == PHP_SORT_REGULAR) {
+		php_array_sort_regular(array, false, false, false);
+		RETURN_TRUE;
+	}
 
+	php_array_apply_sort(array, sort_type, php_get_data_compare_func, false);
 	RETURN_TRUE;
 }
-
-/* {{{ Sort an array and maintain index association */
-PHP_FUNCTION(asort)
-{
-	php_sort(INTERNAL_FUNCTION_PARAM_PASSTHRU, php_get_data_compare_func, false);
-}
 /* }}} */
 
 /* {{{ Sort an array in reverse order and maintain index association */
 PHP_FUNCTION(arsort)
 {
-	php_sort(INTERNAL_FUNCTION_PARAM_PASSTHRU, php_get_data_reverse_compare_func, false);
+	HashTable *array;
+	zend_long sort_type = PHP_SORT_REGULAR;
+
+	ZEND_PARSE_PARAMETERS_START(1, 2)
+		Z_PARAM_ARRAY_HT_EX(array, 0, 1)
+		Z_PARAM_OPTIONAL
+		Z_PARAM_LONG(sort_type)
+	ZEND_PARSE_PARAMETERS_END();
+
+	if ((sort_type & ~PHP_SORT_FLAG_CASE) == PHP_SORT_REGULAR) {
+		php_array_sort_regular(array, false, true, false);
+		RETURN_TRUE;
+	}
+
+	php_array_apply_sort(array, sort_type, php_get_data_reverse_compare_func, false);
+	RETURN_TRUE;
 }
```

Comment:
Please revert these changes, the point of this was to simplify the implementation by having a unique one.
So please change the various `php_get_*_compare_func()`.